modules/images/image-loading-optimization/detection/detect.js

```diff
@@ -84,22 +132,76 @@ function getBreadcrumbs( leafElement ) {
 	return breadcrumbs;
 }
 
+/**
+ * Checks whether the page metric(s) for the provided viewport width is needed.
+ *
+ * @param {number}                   viewportWidth               - Current viewport width.
+ * @param {Array<number, boolean>[]} neededMinimumViewportWidths - Needed minimum viewport widths, in ascending order.
+ * @return {boolean} Whether page metrics are needed.
+ */
+function isViewportNeeded( viewportWidth, neededMinimumViewportWidths ) {
+	let lastWasNeeded = false;
+	for ( const [
+		minimumViewportWidth,
+		isNeeded,
+	] of neededMinimumViewportWidths ) {
+		if ( viewportWidth >= minimumViewportWidth ) {
+			lastWasNeeded = isNeeded;
+		} else {
+			break;
+		}
+	}
+	return lastWasNeeded;
+}
+
+/**
+ * Gets the current time in milliseconds.
+ *
+ * @return {number} Current time in milliseconds.
+ */
+function getCurrentTime() {
+	return new Date().valueOf();
+}
+
 /**
  * Detects the LCP element, loaded images, client viewport and store for future optimizations.
  *
- * @param {number}  serveTime           The serve time of the page in milliseconds from PHP via `ceil( microtime( true ) * 1000 )`.
- * @param {number}  detectionTimeWindow The number of milliseconds between now and when the page was first generated in which detection should proceed.
- * @param {boolean} isDebug             Whether to show debug messages.
+ * @param {Object}                   args                             Args.
+ * @param {number}                   args.serveTime                   The serve time of the page in milliseconds from PHP via `ceil( microtime( true ) * 1000 )`.
+ * @param {number}                   args.detectionTimeWindow         The number of milliseconds between now and when the page was first generated in which detection should proceed.
+ * @param {boolean}                  args.isDebug                     Whether to show debug messages.
+ * @param {string}                   args.restApiEndpoint             URL for where to send the detection data.
+ * @param {string}                   args.restApiNonce                Nonce for writing to the REST API.
+ * @param {string}                   args.pageMetricsSlug             Slug for page metrics.
+ * @param {string}                   args.pageMetricsNonce            Nonce for page metrics storage.
+ * @param {Array<number, boolean>[]} args.neededMinimumViewportWidths Needed minimum viewport widths for page metrics.
+ * @param {number}                   args.storageLockTTL              The TTL (in seconds) for the page metric storage lock.
```

Comment:
The comparison is done in microseconds, so should this value perhaps be provided in microseconds as well?

Reply:
Milliseconds not microseconds, right? As `new Date().valueOf()` returns milliseconds. The server is using `ceil( microtime( true ) * 1000 )` (which is milliseconds). (Though it is also using `time()` which I need to reconcile.)

I thought that seconds for the TTL would be the adequate granularity for developers. I can't imagine a site wanting to filter the TTL to be anything other than whole seconds, but I'm happy to change this.

Reply:
Eh, yes, I meant milliseconds 🤦 

Fair enough 👍 Was just a thought.